waiter/src/waiter/auth/authentication.clj

```diff
@@ -52,23 +52,27 @@
                     {:status http-400-bad-request}))))
 
 (defn- add-cached-auth
-  [response password principal age-in-seconds]
-  (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (tc/to-long (t/now))]
-                                     (or age-in-seconds (-> 1 t/days t/in-seconds))))
+  [response password principal age-in-seconds auth-metadata]
+  (let [cookie-value [principal (tc/to-long (t/now)) auth-metadata]
+        cookie-expiry (or age-in-seconds (-> 1 t/days t/in-seconds))]
+    (cookie-support/add-encoded-cookie
+      response password AUTH-COOKIE-NAME cookie-value cookie-expiry)))
 
 (defn select-auth-params
   "Returns a map that contains only the auth params from the input map"
   [m]
   (select-keys m [:authorization/method :authorization/principal :authorization/user]))
 
-(defn auth-params-map
+(defn build-auth-params-map
```

Comment:
Rename function to avoid confusion with `auth-params-map` variables often used to store the result in callers.